src/com/microsoft/azure/documentdb/JsonSerializable.java

```diff
@@ -440,9 +440,8 @@ void onSave() {
             return c.cast(this.propertyBag);
         } else {
             // POJO
-            if (!c.isMemberClass() || !Modifier.isStatic(c.getModifiers())) {
-                throw new IllegalArgumentException(
-                        "c must be a member (not an anonymous or local) and static class.");
+            if (c.isMemberClass() && !Modifier.isStatic(c.getModifiers())) {
+                throw new IllegalArgumentException("Nested classes must be static.");
```

Comment:
There are similar checks around line 364 too. Can you do me a favor to change the code there? Thanks.